File: 409_longest_palindrome/longestPalindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        seen = set()

        res = 0

        for i in s:
            if i in seen:
                seen.remove(i)
                res += 2
            else:
                seen.add(i)

        if seen:
            res += 1

        return res

File: 409_longest_palindrome/longest_palindrome.rs
use std::collections::HashSet;

impl Solution {
    pub fn longest_palindrome(s: String) -> i32 {
        let mut res = 0;

        let mut visited = HashSet::new();

        for c in s.chars() {
            if visited.contains(&c) {
                res += 2;
                visited.remove(&c);
            } else {
                visited.insert(c);
            }
        }

        if !visited.is_empty() {
            res += 1;
        }

        return res;
    }
}